.github/workflows/main.yml

```diff
@@ -32,6 +32,29 @@ jobs:
         # what images need to be built.
         fetch-depth: 0
 
+    - name: Set up Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: '16'
+
+    # Ideally we'd use `with.cache: 'yarn'` in the `setup-node` action, but that
+    # doesn't support restoring from fallback cache keys in the case of an exact
+    # cache hit: https://github.com/actions/setup-node/issues/328.
```

Comment:
https://github.com/actions/setup-node/issues/328 - there are a few WIP PRs that address this limitation, so hopefully we can switch back to using `actions/setup-node` for caching soon!